### PR TITLE
PLANET-3038 Pass mandatory fields set in an enform block

### PR DESCRIPTION
**Pull request by kirdia**
Pass enform block mandatory fields set in an enform block shortcode to enform twig template.
Change validateForm to fail when at least one of the fields has not passed validation.

### CODE REVIEW

File: assets/js/partials/enform_submit.js [hunk 1]
```diff
@@ -44,9 +44,10 @@ $(document).ready(function () {
 
         formIsValid = false;
       } else if ('email' === $(this).attr('type')) {
-        formIsValid = validateEmail(formValue);
+        var emailValid = validateEmail(formValue);
+        formIsValid = formIsValid === true ? emailValid : false;
 
-        if (!formIsValid) {
+        if (!emailValid) {
```
Comment (angtheod) [inline assets/js/partials/enform_submit.js]: What you think about writing this if/else a bit cleaner, like this
```if (
        $(this).attr('required') && ! formValue ||
        'email' === $(this).attr('type') && ! validateEmail( formValue )
      ) {
        addErrorMessage(this);
        formIsValid = false;
      }```

**Comment by angtheod** [comment]
I know it's not part of the PR but lease change the function name [is_valid_url](https://github.com/greenpeace/planet4-plugin-engagingnetworks/pull/48/files#diff-0737e98aa50baa878050dd5ce03d27cfR111)
to camelCase format, to conform with WP [JS coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/).

**Comment by angtheod** [comment]
What you think about adding keyup event [here](https://github.com/greenpeace/planet4-plugin-engagingnetworks/pull/48/files#diff-0737e98aa50baa878050dd5ce03d27cfR7) so that the errors disappear as soon as user types in a valid value, instead of as soon as input field loses focus (e.g. user clicks elsewhere).

If you agree we would still need to keep both change and keyup events to make sure we also catch the select drop down changes, like this
`$(this).off('change keyup').on('change keyup', function() {`

**Comment by angtheod** [comment]
@pablocubico @comzeradd 
What you think about adding keyup event [here](https://github.com/greenpeace/planet4-plugin-engagingnetworks/pull/48/files#diff-0737e98aa50baa878050dd5ce03d27cfR7) so that the errors disappear as soon as user types in a valid value, instead of as soon as input field loses focus (e.g. user clicks elsewhere).

If you agree we would still need to keep both change and keyup events to make sure we also catch the select drop down changes, like this
`$(this).off('change keyup').on('change keyup', function() {`

**Comment by pablocubico** [comment]
@angtheod Sounds great! Indeed, it should be `keyup` for faster feedback.

**Pull request merged**